### DISCUSSION TITLE
How do you change which Hyper-V virtual switch WSL uses?

### DISCUSSION THREAD

**Discussion by bh2256** [microsoft/WSL#6155] [General]
I noticed in ipconfig that I have two vEthernet adapters, Default Switch and WSL. I confirmed this in Hyper-V Manager - there are two virtual switches. Do I need a separate virtual switch for WSL? Can I just use the Default Switch for WSL and remove the WSL virtual switch? How can I change which virtual switch WSL uses?

Thanks for the help in advance.

**Comment by Biswa96** [comment]
WSL2 virtual network interfaces are created by LxssManager service. It is not documented how it is done. So, fiddling with it may cause your WSL2 instance unusable and machine has to be restarted. In my opinion, "if ain't broke, don't fix it".
